src/shared/CustomInput/CustomInput.tsx

```diff
@@ -63,7 +65,17 @@ export const CustomInput = forwardRef<HTMLSpanElement, Props>(
             parseFloat(inputValue) * parseFloat(assetValue.exchangeRate);
 
         const setMaxAssetAmount = () => {
-            onInputValueChange(balance);
+            if (assetValue.address === TON) {
+                const tonBalanceWithFee =
+                    parseFloat(balance) -
+                    parseFloat(fromNano(GAS_AMOUNT, TON_DECIMALS)) * 4;
```

Comment:
take a look at `rainbow_getTransferParams` you are not considering all gas expenses 
<img width="592" alt="image" src="https://github.com/0xblackbot/rainbow-swap/assets/10281464/5ecb04ca-53f9-429f-ab47-f34bcbd9a0fe">